libsolidity/experimental/analysis/TypeInference.cpp

```diff
@@ -708,11 +713,22 @@ bool TypeInference::visit(TypeClassInstantiation const& _typeClassInstantiation)
 			m_errorReporter.typeError(6948_error, _typeClassInstantiation.location(), "Missing function: " + name);
 			continue;
 		}
+		Type instantiatedClassFunctionType = TypeEnvironmentHelpers{*m_env}.substitute(classFunctionType, classVar, type);
```

Comment:
Ah yeah, ok... this is technically a freshened version of ``classFunctionType`` with the resulting newly introduced type variable unified with ``type``...